### PR TITLE
CCM-3796: Added new proxy endpoint to integrate with NHS app

### PR DESCRIPTION
**Pull request by TheWorldWideWebb**
## Summary
adds nhs proxy endpoint 

## Reviews Required
* [x] Dev
* [ ] Test
* [ ] Tech Author
* [ ] Product Owner

## Checklist
* [ ] Brief description of work completed, and any technical decisions made as part of the PR
* [ ] PR link added as a comment to the relevant JIRA ticket
* [ ] Branch deployed to a dynamic env - link to deployment pipeline
* [ ] PR link shared on Slack and/or Teams
* [ ] 2 reviews received
* [ ] Tester approval

### CODE REVIEW

File: proxies/shared/policies/AssignMessage.NhsAppAccounts.Get.Request.xml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        <Name>requestpath</Name>
+        <Value>/api/channels/nhsapp/accounts</Value>
+    </AssignVariable>
+    <Set>
+        <Headers>
+            <Header name="X-Correlation-Id">{backendCorrelationId}</Header>
+        </Headers>
+        <Verb>GET</Verb>
+        <QueryParams>
+            <QueryParam name="ods-organisation-code">{data.odsOrganisationCode}</QueryParam>
```
Comment (aaron-downing) [inline proxies/shared/policies/AssignMessage.NhsAppAccounts.Get.Request.xml:29]: Is a conversion happening here? Would we not want `data[ods-organisation-code]` instead?

Comment (aaron-downing) [reply proxies/shared/policies/AssignMessage.NhsAppAccounts.Get.Request.xml:29]: Ah, does that mean that the backend will receive the ODS data in a variable called `odsOrganisationCode`?

Comment (simonlabarere) [reply proxies/shared/policies/AssignMessage.NhsAppAccounts.Get.Request.xml:29]: No the backend will get the query param as `ods-organisation-code=Y00001`. `data.odsOrganisationCode` is just where the value is stored in apigee session (whatever it's called).

File: proxies/shared/resources/jsc/NhsAppAccounts.Ok.Response.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+const hostname = virtualhosts.slice(-1)[0];
+const internalHostname = context.getVariable("request.header.host");
+
+const reqUrl = context.getVariable('proxy.url');
+const baseUrl = reqUrl.split("?")[0].replace(internalHostname, hostname);
+
+const parsedContent = JSON.parse(content)
+var odsOrganisationCode = null;
+
+const replacePathRoot = (link) => {
```
Comment (aaron-downing) [inline proxies/shared/resources/jsc/NhsAppAccounts.Ok.Response.js:13]: Could this function be used in `Messages.Create.Response.js`

File: proxies/shared/policies/ExtractVariables.NhsAppAccounts.Get.Request.xml [hunk 3]
```diff
@@ -0,0 +1,18 @@
+<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
+<!--
+ This policy takes the ods-organisation-code value from the URI of the request and stores it within the Apigee flow variable data.ods-organisation-code. This value is thus accessible from other policies in the proxy, referencing it with {data.messageId}.
+
+ For more information, see the following resources:
+    * https://docs.apigee.com/api-platform/fundamentals/introduction-flow-variables
+-->
+<ExtractVariables async="false" continueOnError="false" enabled="true" name="ExtractVariables.NhsAppAccounts.Get.Request">
```
Comment (aaron-downing) [inline proxies/shared/policies/ExtractVariables.NhsAppAccounts.Get.Request.xml:8]: Maybe one for discussion as I'm not sure what the capabilities of APIGEE are and whether we'd want it, but I'm wondering whether we could leverage APIGEE to enforce that ods-organisation-code is a required query param?

File: proxies/shared/resources/jsc/NhsAppAccounts.Ok.Response.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
+const content = context.getVariable("response.content")
+
+const virtualhosts = context.getVariable("virtualhost.aliases");
```
[the remark this answers is not in the document]
Comment (simonlabarere) [reply proxies/shared/resources/jsc/NhsAppAccounts.Ok.Response.js:3]: It looks like it's possible to share javascript files across an environment in Apigee but we'd probably have to check with someone (is there an APIM team?) if they're happy with that.

[the remark this answers is not in the document]
Comment (simonlabarere) [reply proxies/shared/resources/jsc/NhsAppAccounts.Ok.Response.js:3]: Scratch that, it's already done in the repos so it should be easy to do.

File: proxies/shared/policies/ExtractVariables.NhsAppAccounts.Get.Request.xml (hunk 3 above)
Comment (simonlabarere) [reply proxies/shared/policies/ExtractVariables.NhsAppAccounts.Get.Request.xml:8]: Yes that's possible. I'll add it in.

**Review by adadigital — APPROVED** [review]

File: proxies/shared/resources/jsc/NhsAppAccounts.Ok.Response.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (simonlabarere) [reply proxies/shared/resources/jsc/NhsAppAccounts.Ok.Response.js:3]: Common code moved to `responseHelpers.js`

File: proxies/shared/resources/jsc/NhsAppAccounts.Ok.Response.js (hunk 2 above)
Comment (simonlabarere) [reply proxies/shared/resources/jsc/NhsAppAccounts.Ok.Response.js]: Done

File: proxies/shared/policies/ExtractVariables.NhsAppAccounts.Get.Request.xml (hunk 3 above)
Comment (simonlabarere) [reply proxies/shared/policies/ExtractVariables.NhsAppAccounts.Get.Request.xml:8]: Done!

**Review by aaron-downing — APPROVED** [review]

**Pull request merged**